tower_cli/resources/job_template.py

```diff
@@ -145,3 +146,18 @@ def disassociate_notification_template(self, job_template,
         """Disassociate a notification template from this job template."""
         return self._disassoc('notification_templates_%s' % status,
                               job_template, notification_template)
+
+    @resources.command(use_fields_as_options=('extra_vars'))
+    @click.option('--host-config-key', help='Job-template-specific string used to authenticate '
+                  'host during provisioning callback.')
+    def callback(self, pk=None, host_config_key='', extra_vars=None):
+        """Contact Tower and request a configuration update using this job template."""
+        url = self.endpoint + '%s/callback/' % pk
+        if not host_config_key:
+            host_config_key = client.get(url).json()['host_config_key']
```

Comment:
so the big question here is if this authentication _by itself_ will be sufficient to authenticate the request, even if the tower-cli config is wrong. Docs:

http://docs.ansible.com/ansible-tower/2.2.0/html/userguide/job_templates.html#provisioning-callbacks

```bash
root@localhost:~$ curl --data "host_config_key=5a8ec154832b780b9bdef1061764ae5a" \
                    http:///api/v1/job_templates/1/callback/
```

This seems to suggest that it would. Even if (wrong) basic auth parameters are given along with it, I would put my money on it working.